### PR TITLE
[e2e] refactor POM classes structure

### PR DESCRIPTION
**Pull request by anezboretskiy**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [x] No need to
- [ ] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaRegistryList.java [hunk 1]
```diff
@@ -3,40 +3,41 @@
...
 
     @Step
     public SchemaCreateForm clickCreateSchema() {
-        clickByJavaScript(schemaButton);
+        clickByJavaScript(createSchemaBtn);
         return new SchemaCreateForm();
     }
 
     @Step
     public SchemaDetails openSchema(String schemaName) {
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaRegistryList.java:32]: pls change the method to return SchemaCreateForm

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaRegistryList.java [hunk 2]
```diff
@@ -3,40 +3,41 @@
...
     public SchemaRegistryList waitUntilScreenReady(){
-        $x("//*[contains(text(),'Loading')]").shouldBe(Condition.disappear);
-        $x("//button[contains(text(),'Create Schema')]").shouldBe(Condition.visible);
+        loadingSpinner.shouldBe(Condition.disappear);
+        createSchemaBtn.shouldBe(Condition.visible);
         return this;
     }
 
     @Step
     public SchemaCreateForm clickCreateSchema() {
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaRegistryList.java:26]: pls change the method to return SchemaCreateForm

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java [hunk 3]
```diff
@@ -5,26 +5,38 @@
...
 import static com.codeborne.selenide.Selenide.*;
 import static com.provectus.kafka.ui.utilities.WebUtils.clickByJavaScript;
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class TopicCreateEditForm {
 
-    private final SelenideElement timeToRetain = $(By.cssSelector("input#timeToRetain"));
-    private final SelenideElement maxMessageBytes = $(By.name("maxMessageBytes"));
+    protected SelenideElement loadingSpinner = $x("//*[contains(text(),'Loading')]");
+    protected SelenideElement timeToRetain = $x("//input[@id='timeToRetain']");
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java:16]: fld?

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java [hunk 4]
```diff
@@ -5,26 +5,38 @@
...
+    protected SelenideElement inputNameField = $x("//input[@name='name']");
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java:17]: please don't use 'input'

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java [hunk 5]
```diff
@@ -5,26 +5,38 @@
...
+    protected SelenideElement maxMessageBytes = $x("//input[@name='maxMessageBytes']");
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java:18]: fld??

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java [hunk 6]
```diff
@@ -5,26 +5,38 @@
...
+    protected SelenideElement minInSyncReplicas = $x("//input[@name='minInSyncReplicas']");
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java:19]: fld???

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java [hunk 7]
```diff
@@ -5,26 +5,38 @@
...
+    protected SelenideElement cleanUpPolicyDropDown = $x("//ul[@id='topicFormCleanupPolicy']");
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java:20]: Ddl pls

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicDetails.java [hunk 8]
```diff
@@ -7,24 +7,34 @@
...
+
 import static com.codeborne.selenide.Selenide.*;
 import static com.provectus.kafka.ui.utilities.WebUtils.clickByJavaScript;
 
 @ExtensionMethod({WaitUtils.class})
 public class TopicDetails {
 
+    protected SelenideElement loadingSpinner = $x("//*[contains(text(),'Loading')]");
     protected SelenideElement dotMenuBtn = $$x("//button[@aria-label='Dropdown Toggle']").first();
+    protected List<SelenideElement> navigationTabs = $$x("(//nav[@role='navigation'])[last()]/a");
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicDetails.java:20]: use Arrays.asList(bla-bla to use loop just in one place

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicDetails.java [hunk 9]
```diff
@@ -37,16 +47,15 @@ public TopicDetails openTopicMenu(TopicMenu menu) {
     @Step
     public TopicsList deleteTopic() {
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicDetails.java:48]: make method return TopicDetails

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicDetails.java [hunk 10]
```diff
@@ -37,16 +47,15 @@ public TopicDetails openTopicMenu(TopicMenu menu) {
...
+        removeTopicBtn.shouldBe(Condition.visible).click();
+        confirmBtn.shouldBe(Condition.enabled).click();
+        confirmBtn.shouldBe(Condition.disappear);
         return new TopicsList();
     }
 
     @Step
-    public ProduceMessagePanel clickOnButton(String buttonName) {
-        clickByJavaScript($(By.xpath(String.format("//div//button[text()='%s']", buttonName))));
+    public ProduceMessagePanel openProduceMessagePanel() {
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicDetails.java:57]: make method return TopicDetails

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicsList.java [hunk 11]
```diff
@@ -14,23 +15,29 @@
...
-        $(By.xpath("//h1[text()='Topics']")).shouldBe(Condition.visible);
+        loadingSpinner.shouldBe(Condition.disappear);
+        topicListHeader.shouldBe(Condition.visible);
         return this;
     }
 
     @Step
-    public TopicCreateEditForm pressCreateNewTopic() {
-        clickByJavaScript($x("//button[normalize-space(text()) ='Add a Topic']"));
+    public TopicCreateEditForm clickCreateNewTopic() {
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicsList.java:32]: make method return TopicsList

**Review by VladSenyuta — CHANGES_REQUESTED** [review]

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorCreateForm.java [hunk 12]
```diff
@@ -3,37 +3,37 @@
...
 import static com.provectus.kafka.ui.utilities.WebUtils.clickByJavaScript;
 import static com.provectus.kafka.ui.utilities.screenshots.Screenshooter.log;
 
 public class ConnectorCreateForm {
 
-    protected SelenideElement nameField = $(By.xpath("//input[@name='name']"));
-    protected SelenideElement contentTextArea = $(".ace_text-input");
-    protected SelenideElement submitButton = $(By.xpath("//button[@type='submit']"));
+    protected SelenideElement loadingSpinner = $x("//*[contains(text(),'Loading')]");
+    protected SelenideElement inputNameField = $x("//input[@name='name']");
```
[the remark this answers is not in the document]
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorCreateForm.java:15]: Refactored to nameField

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorDetails.java [hunk 13]
```diff
@@ -3,41 +3,51 @@
...
+    protected SelenideElement dotMenuBtn = $x("//button[@aria-label='Dropdown Toggle']");
+    protected SelenideElement deleteBtn = $x("//li/div[contains(text(),'Delete')]");
+    protected SelenideElement confirmBtnMdl = $x("//div[@role='dialog']//button[contains(text(),'Confirm')]");
+    protected SelenideElement submitBtn = $x("//button[@type='submit']");
+    protected SelenideElement contentTextArea = $x("//textarea[@class='ace_text-input']");
+    protected SelenideElement taskTab = $x("//a[contains(text(),'Tasks')]");
+    protected SelenideElement configTab = $x("//a[contains(text(),'Config')]");
+    protected SelenideElement overviewTab = $x("//a[contains(text(),'Overview')]");
+    protected SelenideElement configField = $x("//div[@id='config']");
+    protected List<SelenideElement> navigationElements = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorDetails.java:28]: Deleted and using Arrayys.asList

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorDetails.java [hunk 14]
```diff
@@ -3,41 +3,51 @@
...
 
     @Step
     public ConnectorDetails waitUntilScreenReady() {
-        $(By.xpath("//a[text() ='Tasks']")).shouldBe(Condition.visible);
-        $(By.xpath("//a[text() ='Config']")).shouldBe(Condition.visible);
+        loadingSpinner.shouldBe(Condition.disappear);
+        navigationElements.addAll(List.of(taskTab, configTab, overviewTab));
```
[the remark this answers is not in the document]
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorDetails.java:33]: Done

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorDetails.java [hunk 15]
```diff
@@ -3,41 +3,51 @@
...
     public ConnectorDetails openConfigTab() {
-        clickByJavaScript($(By.xpath("//a[text() ='Config']")));
+        clickByJavaScript(configTab);
         return this;
     }
 
     @Step()
     public ConnectorDetails setConfig(String configJson) {
-        $("#config").click();
+        configField.shouldBe(Condition.visible).click();
```
[the remark this answers is not in the document]
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorDetails.java:46]: Done

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/KafkaConnectList.java [hunk 16]
```diff
@@ -14,34 +15,41 @@
 @ExtensionMethod(WaitUtils.class)
 public class KafkaConnectList {
 
+    protected SelenideElement loadingSpinner = $x("//*[contains(text(),'Loading')]");
+    protected SelenideElement headerConnectors = $x("//h1[text()='Connectors']");
```
[the remark this answers is not in the document]
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/KafkaConnectList.java:19]: Refactored to titleConnectors

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaCreateForm.java [hunk 17]
```diff
@@ -15,15 +14,22 @@
 
 public class SchemaCreateForm {
 
-    protected SelenideElement subjectName = $(By.xpath("//input[@name='subject']"));
-    protected SelenideElement schemaField = $(By.xpath("//textarea[@name='schema']"));
-    protected SelenideElement submitSchemaButton = $(By.xpath("//button[@type='submit']"));
+    protected SelenideElement loadingSpinner = $x("//*[contains(text(),'Loading')]");
+    protected SelenideElement subjectName = $x("//input[@name='subject']");
+    protected SelenideElement titleElementLocator = $x("//h1['Edit']");
```
[the remark this answers is not in the document]
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaCreateForm.java:19]: Refactored

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaCreateForm.java [hunk 18]
```diff
@@ -15,15 +14,22 @@
...
+    protected SelenideElement schemaField = $x("//textarea[@name='schema']");
+    protected SelenideElement submitSchemaButton = $x("//button[@type='submit']");
     protected SelenideElement newSchemaTextArea = $("#newSchema [wrap]");
     protected SelenideElement schemaTypeDropDown = $x("//ul[@name='schemaType']");
+    protected SelenideElement schemaTypeList = $x("//ul[@role='listbox']");
```
[the remark this answers is not in the document]
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaCreateForm.java:24]: Refactored to ddlSchemaType = $x("//ul[@name='schemaType']")

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaDetails.java [hunk 19]
```diff
@@ -3,39 +3,45 @@
...
+import static com.codeborne.selenide.Selenide.$$x;
+import static com.codeborne.selenide.Selenide.$x;
 import static com.provectus.kafka.ui.utilities.WebUtils.clickByJavaScript;
 
 public class SchemaDetails {
 
     protected SelenideElement dotMenuBtn = $$x("//button[@aria-label='Dropdown Toggle']").first();
+    protected SelenideElement loadingSpinner = $x("//*[contains(text(),'Loading')]");
+    protected SelenideElement actualVersionWindow = $x("//div[@id='schema']");
+    protected SelenideElement compatibility = $x("//h4[contains(text(),'Compatibility')]/../p");
```
[the remark this answers is not in the document]
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaDetails.java:16]: changed according to types

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaDetails.java [hunk 20]
```diff
@@ -3,39 +3,45 @@
...
-import static com.codeborne.selenide.Selenide.*;
+import static com.codeborne.selenide.Selenide.$$x;
+import static com.codeborne.selenide.Selenide.$x;
 import static com.provectus.kafka.ui.utilities.WebUtils.clickByJavaScript;
 
 public class SchemaDetails {
 
     protected SelenideElement dotMenuBtn = $$x("//button[@aria-label='Dropdown Toggle']").first();
+    protected SelenideElement loadingSpinner = $x("//*[contains(text(),'Loading')]");
+    protected SelenideElement actualVersionWindow = $x("//div[@id='schema']");
```
[the remark this answers is not in the document]
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaDetails.java:15]: Refactored to actualVersionMdlWindow

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/ProduceMessagePanel.java [hunk 21]
```diff
@@ -6,15 +6,25 @@
...
-    private final SelenideElement keyField = $(By.xpath("//div[@id='key']/textarea"));
-    private final SelenideElement contentField = $(By.xpath("//div[@id='content']/textarea"));
-    private final SelenideElement headersField = $(By.xpath("//div[@id='headers']/textarea"));
-    private final SelenideElement submitBtn = headersField.$(By.xpath("../../../..//button[@type='submit']"));
+    protected SelenideElement loadingSpinner = $x("//*[contains(text(),'Loading')]");
+    protected SelenideElement keyField = $(By.xpath("//div[@id='key']/textarea"));
+    protected SelenideElement contentField = $(By.xpath("//div[@id='content']/textarea"));
+    protected SelenideElement headersField = $(By.xpath("//div[@id='headers']/textarea"));
+    protected SelenideElement submitBtn = headersField.$(By.xpath("../../../..//button[@type='submit']"));
+    protected List<SelenideElement> ddElementsLocator = $$x("//ul[@role='listbox']");
```
[the remark this answers is not in the document]
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/ProduceMessagePanel.java:20]: Refactored to ddlElementsLocator

[the remark this answers is not in the document]
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/ProduceMessagePanel.java:20]: It is only ddl Elements like this , it is not title 
Screenshot_4

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java (hunk 3 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java:16]: Refactored to timeToRetainField

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java (hunk 4 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java:17]: Refactored to nameField

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java (hunk 5 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java:18]: Refactored to +Field

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java (hunk 6 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java:19]: Refactored to +Field

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicDetails.java (hunk 8 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicDetails.java:20]: Refactored to using Arrays.asList

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicDetails.java (hunk 9 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicDetails.java:48]: Done

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicDetails.java (hunk 10 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicDetails.java:57]: Done

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicsList.java (hunk 11 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicsList.java:32]: Done

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaRegistryList.java (hunk 2 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaRegistryList.java:26]: Refactored to return SchemaRegistryList

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/ProduceMessagePanel.java (hunk 21 above)
[the remark this answers is not in the document]
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/ProduceMessagePanel.java:20]: Refactored to individual SelenideElements

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/KafkaConnectList.java (hunk 16 above)
[the remark this answers is not in the document]
Comment (VladSenyuta) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/KafkaConnectList.java]: as we always use: name of element than type of element -> connectorsTitle. but I suggest to use pageTitle in this case

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaCreateForm.java (hunk 18 above)
[the remark this answers is not in the document]
Comment (VladSenyuta) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaCreateForm.java]: as we always use: name of element than type of element -> schemaTypeDdl

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java [hunk 22]
```diff
@@ -93,9 +104,9 @@ public TopicCreateEditForm selectRetentionBytes(Long optionValue) {
     }
 
     @Step
-    public TopicDetails sendData() {
-        clickByJavaScript($x("//button[@type='submit']"));
-        return new TopicDetails();
+    public TopicCreateEditForm submitCreateTopicBtn() {
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java:107]: this method does click on button. why 'submit'??

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicDetails.java [hunk 23]
```diff
@@ -35,19 +46,18 @@ public TopicDetails openTopicMenu(TopicMenu menu) {
...
+        confirmBtn.shouldBe(Condition.enabled).click();
+        confirmBtn.shouldBe(Condition.disappear);
+        return this;
     }
 
     @Step
-    public ProduceMessagePanel clickOnButton(String buttonName) {
-        clickByJavaScript($(By.xpath(String.format("//div//button[text()='%s']", buttonName))));
-        return new ProduceMessagePanel();
+    public TopicDetails openProduceMessagePanel() {
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicDetails.java:58]: let's use literally meanings og our actions. clickProduceMessageButton(), pls

**Review by VladSenyuta — CHANGES_REQUESTED** [review]

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicDetails.java (hunk 23 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicDetails.java:58]: renamed

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java (hunk 22 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/TopicCreateEditForm.java:107]: Renamed

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/KafkaConnectList.java (hunk 16 above)
[the remark this answers is not in the document]
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/KafkaConnectList.java]: Refactored to pageTitle

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaCreateForm.java (hunk 18 above)
[the remark this answers is not in the document]
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaCreateForm.java]: Refactored according to specs

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorCreateForm.java [hunk 24]
```diff
@@ -3,37 +3,37 @@
...
+        return this;
+    }
 
     @Step("Set connector config JSON")
-    public ConnectorDetails setConnectorConfig(String connectName, String configJson) {
-        nameField.setValue(connectName);
-        $("#config").click();
+    public ConnectorCreateForm setConnectorConfig(String connectName, String configJson) {
+        nameField.shouldBe(Condition.enabled).setValue(connectName);
+        configField.shouldBe(Condition.visible).click();
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorCreateForm.java:30]: I believe we can use 'enabled' condition here

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorCreateForm.java [hunk 25]
```diff
@@ -3,37 +3,37 @@
...
         contentTextArea.setValue("");
         contentTextArea.setValue(String.valueOf(configJson.toCharArray()));
-        nameField.click();
-        clickByJavaScript(submitButton);
+        nameField.shouldBe(Condition.visible).click();
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorCreateForm.java:33]: I believe we can use 'enabled' condition here

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/ProduceMessagePanel.java [hunk 26]
```diff
@@ -3,46 +3,58 @@
...
+        loadingSpinner.shouldBe(Condition.disappear);
+        Arrays.asList(partitionDdl, keySerdeDdl, contentSerdeDdl).forEach(element -> element.shouldBe(Condition.visible));
+        return this;
+    }
 
     @Step
     public ProduceMessagePanel setKeyField(String value) {
-        keyField.shouldBe(Condition.enabled)
+        keyTextArea.shouldBe(Condition.enabled)
                 .sendKeys(Keys.chord(Keys.DELETE));
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/ProduceMessagePanel.java:34]: why do we use 'Keys.chord' here?

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaDetails.java (hunk 20 above)
[the remark this answers is not in the document]
Comment (VladSenyuta) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaDetails.java]: I have a concern about the type here, let's discuss

**Review by VladSenyuta — CHANGES_REQUESTED** [review]

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorCreateForm.java (hunk 24 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorCreateForm.java:30]: changed to enabled

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorCreateForm.java (hunk 25 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorCreateForm.java:33]: changed to enabled

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/ProduceMessagePanel.java (hunk 26 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/topic/ProduceMessagePanel.java:34]: Refactored to clearByKeyboard()

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaDetails.java (hunk 20 above)
[the remark this answers is not in the document]
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaDetails.java]: Refactored to actualVersionTextArea

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorDetails.java [hunk 27]
```diff
@@ -3,41 +3,47 @@
...
         return this;
     }
 
     @Step()
     public ConnectorDetails setConfig(String configJson) {
-        $("#config").click();
-        contentTextArea.sendKeys(Keys.LEFT_CONTROL + "a");
+        configField.shouldBe(Condition.enabled).click();
+        clearByKeyboard(contentTextArea);
         contentTextArea.setValue("");
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorDetails.java:44]: can we simplify someway here?

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaCreateForm.java [hunk 28]
```diff
@@ -6,73 +6,78 @@
...
         return this;
     }
 
     @Step("Set new schema value")
     public SchemaCreateForm setNewSchemaValue(String configJson) {
-        $("#newSchema").click();
-        newSchemaTextArea.sendKeys(Keys.CONTROL + "a", Keys.BACK_SPACE);
+        newSchemaTextArea.shouldBe(Condition.visible).click();
+        clearByKeyboard(newSchemaInput);
         Selenide.executeJavaScript("arguments[0].value = '';", $("#newSchema"));
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaCreateForm.java:71]: can we simplify someway here?

**Review by VladSenyuta — CHANGES_REQUESTED** [review]

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorDetails.java (hunk 27 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorDetails.java:44]: Reformat without 
contentTextArea.setValue("");
        contentTextArea.setValue(String.valueOf(configJson.toCharArray()));

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaCreateForm.java (hunk 28 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaCreateForm.java:71]: simplify with clickByJavaScript() method

Comment (VladSenyuta) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaCreateForm.java]: what??

Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/schema/SchemaCreateForm.java]: Deleted clickByJavaScript() , cause we cleared field in clearByKeyboard. JS injection here impractical

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorDetails.java [hunk 29]
```diff
@@ -3,41 +3,46 @@
...
-        contentTextArea.sendKeys(Keys.LEFT_CONTROL + "a");
-        contentTextArea.setValue("");
-        contentTextArea.setValue(String.valueOf(configJson.toCharArray()));
-        $("#config").click();
+        configField.shouldBe(Condition.enabled).click();
+        clearByKeyboard(contentTextArea);
+        contentTextArea.setValue(configJson);
+        configField.shouldBe(Condition.enabled).click();
         clickByJavaScript(submitBtn);
         sleep(4000);
```
Comment (VladSenyuta) [inline kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorDetails.java:47]: can we do smth with this hardcoded wait?

**Review by VladSenyuta — CHANGES_REQUESTED** [review]

File: kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorDetails.java (hunk 29 above)
Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorDetails.java:47]: sure , deleted

Comment (anezboretskiy) [reply kafka-ui-e2e-checks/src/main/java/com/provectus/kafka/ui/pages/connector/ConnectorDetails.java]: And added alertMessage condition

**Review by VladSenyuta — APPROVED** [review]

**Pull request merged**